src/frontend.md

```diff
@@ -499,6 +499,84 @@ export function fakeTicket() {}
    const documentMutation = usePatchDocument(documentId);
    ```
 
+## I18n
+
+### Translation keys
+
+1. Use `snake_case` for translation keys.
+
+   ```typescript
+   // bad
+   t('home.helloWorld');
+
+   // good
+   t('home.hello_world');
+   ```
+
+1. Do not split phrases into several translation keys. Sometimes it's impossible to translate splitted phrase to another language.
+
+   ```typescript
+   // bad
+   const greeting = t('home.hello') + userName + '!';
+
+   // good
+   // in translation file: "hello_user": "Hello, {{userName}}!",
+   const greeting = t('home.hello_user', { userName });
+   ```
+
+1. Give meaningful names to translation keys. Translation key should shortly describe their purpose of content.
+
+   ```typescript
+   // bad
+   t('home.header_title');
+
+   // good
+   t('home.hello_world');
+   ```
+
+1. Do not nest translation keys too deeply, keep 2-3 levels of nesting.
```

Comment:
Do we have a rule – how to manage with levels?

`invoices_payable.title_invoices_to_pay` - what is "invoices_payable" here? The name of the component?

Also, please provide a good example with 3 levels.

Reply:
Updated this rule